pkg/harvester/list/harvesterhci.io.host.vue

```diff
@@ -62,8 +62,42 @@ export default {
       _hash.machines = this.$store.dispatch(`${ inStore }/findAll`, { type: CAPI.MACHINE });
     }
 
+    if (
+      this.$store.getters['rancher/schemaFor'](NORMAN.PRINCIPAL) &&
+      this.$store.getters['rancher/schemaFor'](NORMAN.CLUSTER_ROLE_TEMPLATE_BINDING)
+    ) {
+      _hash.normanPrincipal = this.$store.dispatch('rancher/findAll', { type: NORMAN.PRINCIPAL });
+      _hash.clusterRoleTemplateBinding = this.$store.dispatch(`management/findAll`, { type: MANAGEMENT.CLUSTER_ROLE_TEMPLATE_BINDING });
+    }
+
     const hash = await allHash(_hash);
 
+    // Remove delete action if current user role is cluster member
+    if (hash.normanPrincipal && hash.clusterRoleTemplateBinding) {
+      const role = hash.clusterRoleTemplateBinding.find(
+        (template) => template.userPrincipalName === hash.normanPrincipal[0]?.id
+      );
+      const isClusterMember = role?.roleTemplateName === 'cluster-member';
+
+      if (isClusterMember) {
+        hash.nodes = hash.nodes.map((node) => {
+          const updatedActions = node.availableActions.map((action) => {
+            return action.action === 'promptRemove' ? { ...action, enabled: false } : action;
+          });
+
+          // keep availableActions non-enumerable
+          Object.defineProperty(node, 'availableActions', {
```

Comment:
In general, `availableActions` should be defined in the model, in this case it would be `pkg/harvester/models/harvester/node.js`.

Reply:
Sounds good, but the trade-off is that the model will have a broader impact, and using async within the getter could lead to infinite recursion, so we need the `_initialized` flag to prevent repetition.
update in commit https://github.com/harvester/harvester-ui-extension/pull/83/commits/43b10b250ef1dc89982a54677c53de040b009f4d.